src/pkg/shub/client/api.go

```diff
@@ -78,6 +78,9 @@ func getManifest(uri ShubURI) (manifest ShubAPIResponse, err error) {
 
 	// Do the request, if status isn't success, return error
 	res, err := httpc.Do(req)
+	if res == nil {
+		return ShubAPIResponse{}, fmt.Errorf("No response received from singularity hub")
+	}
```

Comment:
```go
	if resp.StatusCode == http.StatusNotFound {
		return fmt.Errorf("The requested image was not found in singularity hub")
	}
```
as suggested by @tri-adam

Reply:
That check is already there, I think he was saying that not having this direct check of `res == nil` and just going straight into a `res.StatusCode` is what's causing the panic. @tri-adam correct me if im wrong though

Reply:
sorry, didn't read it all. my bad

Reply:
no worries